src/core_modules/capture-core/components/WidgetEventEdit/EditEventDataEntry/editEventDataEntry.epics.js

```diff
@@ -190,3 +193,23 @@ export const requestDeleteEventDataEntryEpic = (action$: InputObservable, store:
             return startDeleteEventDataEntry(eventId, params);
         }));
 
+export const startCreateNewAfterCompletingEpic = (
+    action$: InputObservable, store: ReduxStore, { history }: ApiUtils) =>
+    action$.pipe(
+        ofType(
+            actionTypes.START_CREATE_NEW_AFTER_COMPLETING,
+            newEventWidgetActionTypes.START_CREATE_NEW_AFTER_COMPLETING,
+        ),
+        flatMap((action) => {
+            const { isCreateNew, enrollmentId, orgUnitId, programId, teiId, availableProgramStages } = action.payload;
+            const params = { enrollmentId, orgUnitId, programId, teiId };
+
+            if (isCreateNew) {
+                const finalParams = availableProgramStages.length === 1 ?
+                    { ...params, stageId: availableProgramStages[0].id } : params;
+                history.push(`/enrollmentEventNew?${buildUrlQueryString(finalParams)}`);
+                return EMPTY;
+            }
+            return of(navigateToEnrollmentOverview(params));
```

Comment:
NIT: 
As far as I am aware, the user has to go through the enrollment dashboard to reach the pages where it can add/edit an enrollment event. Therefore we can assume that the admin has already opted-in to use the enrollment dashboard for that program.
Based on this, I believe there is no need to check where to navigate, and is safe to go directly to the enrollment page like this: 
```
    history.push(`/enrollment?${buildUrlQueryString(params)}`);
    return EMPTY;
```